pkg/controller/certificate-shim/sync.go

```diff
@@ -138,22 +140,26 @@ func SyncFnFor(
 
 		for _, crt := range updateCrts {
 
+			obj := &cmapi.Certificate{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            crt.Name,
+					Namespace:       crt.Namespace,
+					Labels:          crt.Labels,
+					OwnerReferences: crt.OwnerReferences,
+				},
+				Spec: cmapi.CertificateSpec{
+					DNSNames:    crt.Spec.DNSNames,
+					IPAddresses: crt.Spec.IPAddresses,
+					IssuerRef:   crt.Spec.IssuerRef,
+					Usages:      crt.Spec.Usages,
+				},
+			}
+			if len(extraAnnotations) > 0 {
+				obj.ObjectMeta.Annotations = extraAnnotations
+			}
+
 			if utilfeature.DefaultFeatureGate.Enabled(feature.ServerSideApply) {
-				err = internalcertificates.Apply(ctx, cmClient, fieldManager, &cmapi.Certificate{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:            crt.Name,
-						Namespace:       crt.Namespace,
-						Labels:          crt.Labels,
-						OwnerReferences: crt.OwnerReferences,
-					},
-					Spec: cmapi.CertificateSpec{
-						DNSNames:    crt.Spec.DNSNames,
-						IPAddresses: crt.Spec.IPAddresses,
-						SecretName:  crt.Spec.SecretName,
-						IssuerRef:   crt.Spec.IssuerRef,
-						Usages:      crt.Spec.Usages,
-					},
-				})
+				err = internalcertificates.Apply(ctx, cmClient, fieldManager, obj)
```

Comment:
Paging @erikgb for a check that this is safe; I don't trust myself when it comes to server side apply!

Reply:
Why will it break? You can just init the annotations variable as `nil`, or?

Reply:
@erikgb I've pushed new commit, could you please if you like that approach better

Reply:
With the alternative approach, the `extractExtraAnnotations` function must return `nil` if there are no annotations to add. But maybe a conditional for empty map in the SSA section is better?

Reply:
```
But maybe a conditional for the empty map in the SSA section is better?
```
I believe that was my initial approach

Reply:
The SSA code looks good to me now. 🚀 And I think @SgtCoDFish has plans to review this PR again later today.

Reply:
resolving conversation based [on](https://kubernetes.slack.com/archives/CDEQJ0Q8M/p1742555009716529?thread_ts=1740410760.670419&cid=CDEQJ0Q8M)